modules/core/src/lib/attribute/attribute-manager.js

```diff
@@ -27,35 +27,26 @@ import AttributeTransitionManager from './attribute-transition-manager';
 const LOG_START_END_PRIORITY = 2;
 const LOG_DETAIL_PRIORITY = 3;
 
-function noop() {}
-
 // Default loggers
 const logFunctions = {
-  savedMessages: null,
+  savedMessages: [],
   timeStart: null,
-  onLog: ({level, message}) => {
-    log.log(level, message)();
-  },
-  onUpdateStart: ({level, numInstances}) => {
-    logFunctions.savedMessages = [];
+  onUpdateStart: () => {
+    logFunctions.savedMessages.length = 0;
     logFunctions.timeStart = new Date();
   },
-  onUpdate: ({level, message}) => {
-    if (logFunctions.savedMessages) {
-      logFunctions.savedMessages.push(message);
-    }
+  onUpdate: message => {
+    logFunctions.savedMessages.push(message);
   },
-  onUpdateEnd: ({level, id, numInstances}) => {
```

Comment:
Is this all internal? Do any docs need to be updated?

Reply:
This class is internal.